grafana/dashboard.json

```diff
@@ -0,0 +1,1714 @@
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 45
+      },
+      "id": 10,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
+        },
+        "tooltip": {
+          "mode": "single",
+          "sort": "none"
+        }
+      },
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "PBFA97CFB590B2093"
+          },
+          "disableTextWrap": false,
+          "editorMode": "builder",
+          "expr": "cyclops_cache_keys_added",
+          "fullMetaSearch": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "legendFormat": "{{pod}} - added",
+          "range": true,
+          "refId": "A",
+          "useBackend": false
+        },
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "PBFA97CFB590B2093"
+          },
+          "disableTextWrap": false,
+          "editorMode": "builder",
+          "expr": "cyclops_cache_keys_evicted",
+          "fullMetaSearch": false,
+          "hide": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "legendFormat": "{{pod}} - evicted",
+          "range": true,
+          "refId": "B",
+          "useBackend": false
+        }
+      ],
+      "title": "cache keys added/evicted",
+      "type": "timeseries"
+    }
+  ],
+  "refresh": "30s",
+  "schemaVersion": 39,
+  "tags": [],
+  "templating": {
+    "list": [
+      {
+        "allValue": ".*",
+        "current": {
+          "selected": true,
+          "text": [
+            "cyclops"
+          ],
+          "value": [
+            "cyclops"
+          ]
+        },
+        "datasource": {
+          "type": "prometheus",
+          "uid": "PBFA97CFB590B2093"
+        },
+        "definition": "",
+        "hide": 0,
+        "includeAll": true,
+        "multi": true,
+        "name": "namespace",
+        "options": [],
+        "query": "label_values(go_memstats_alloc_bytes, namespace)",
+        "refresh": 2,
+        "regex": "",
+        "skipUrlSync": false,
+        "sort": 0,
+        "tagValuesQuery": "",
+        "tagsQuery": "",
+        "type": "query",
+        "useTags": false
+      },
+      {
+        "allValue": ".*",
+        "current": {
+          "selected": true,
+          "text": [
+            "cyclops-ctrl-57846855dc-ls9v2"
+          ],
+          "value": [
+            "cyclops-ctrl-57846855dc-ls9v2"
+          ]
+        },
+        "datasource": {
+          "type": "prometheus",
+          "uid": "PBFA97CFB590B2093"
+        },
+        "definition": "",
+        "hide": 0,
+        "includeAll": true,
+        "multi": true,
+        "name": "pod",
+        "options": [],
+        "query": "label_values(process_resident_memory_bytes, pod)",
```

Comment:
Can we limit the pods only to the ones from the specified `$namespace` and that emit cyclops specific metrics
```suggestion
        "definition": "label_values(cyclops_modules_deployed{namespace=\"$namespace\"},pod)",
        "hide": 0,
        "includeAll": true,
        "multi": true,
        "name": "pod",
        "options": [],
        "query": {
          "qryType": 1,
          "query": "label_values(cyclops_modules_deployed{namespace=\"$namespace\"},pod)",
          "refId": "PrometheusVariableQueryEditor-VariableQuery"
        },
```

Reply:
Do you mean in pod filtering?
![image](https://github.com/user-attachments/assets/59e66944-42f6-42d6-8f79-4468e34a2fed)

Reply:
@rimvydascivilis yeah. I would expect to see only cyclops-ctrl pods in that select. With the query:
```
label_values(cyclops_modules_deployed{namespace=\"$namespace\"},pod)
```
we are getting all pods that emit metric `cyclops_modules_deployed` in a namespace that was selected in the namespace filter